File: sources/nitter-proxy.py
#!/usr/bin/env python3
from flask import Flask, request, Response
import requests
import re

app = Flask(__name__)

@app.route('/proxy')
def proxy():
    username = request.args.get('username', '')
    
    if not username:
        return 'Username required', 400
    
    # Sanitize username
    if not re.match(r'^[a-zA-Z0-9_]+$', username):
        return 'Invalid username', 400
    
    url = f'https://nitter.net/{username}'
    
    try:
        response = requests.get(
            url,
            headers={'User-Agent': 'Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36'},
            timeout=30
        )
        
        return Response(
            response.content,
            status=response.status_code,
            headers={
                'Content-Type': 'text/html; charset=utf-8',
                'Access-Control-Allow-Origin': '*'
            }
        )
    except Exception as e:
        return f'Error: {str(e)}', 500

if __name__ == '__main__':
    app.run(host='0.0.0.0', port=5000)

File: sources/nitter-proxy.php
<?php
header('Content-Type: text/html; charset=utf-8');
header('Access-Control-Allow-Origin: *');

$username = isset($_GET['username']) ? $_GET['username'] : '';

if (empty($username)) {
    http_response_code(400);
    echo 'Username required';
    exit;
}

// Sanitize username (only allow alphanumeric and underscore)
if (!preg_match('/^[a-zA-Z0-9_]+$/', $username)) {
    http_response_code(400);
    echo 'Invalid username';
    exit;
}

$url = 'https://nitter.net/' . $username;

$ch = curl_init();
curl_setopt($ch, CURLOPT_URL, $url);
curl_setopt($ch, CURLOPT_RETURNTRANSFER, true);
curl_setopt($ch, CURLOPT_FOLLOWLOCATION, true);
curl_setopt($ch, CURLOPT_USERAGENT, 'Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36');
curl_setopt($ch, CURLOPT_TIMEOUT, 30);

$response = curl_exec($ch);
$httpCode = curl_getinfo($ch, CURLINFO_HTTP_CODE);

if (curl_errno($ch)) {
    http_response_code(500);
    echo 'Error: ' . curl_error($ch);
    curl_close($ch);
    exit;
}

curl_close($ch);

http_response_code($httpCode);
echo $response;
?>
